docs/thinking_in_jax.rst

```diff
@@ -0,0 +1,484 @@
+How to think in JAX
+===================
+
+
+JAX provides a simple and powerful API for writing accelerated numerical code,
+but working effectively in JAX sometimes requires extra consideration.
+This document is meant to help build a ground-up understanding of how JAX operates,
+so that you can use it more effectively.
+
+JAX vs. Numpy
+-------------
+**Key Concepts:**
+
+- JAX provides a numpy-inspired interface for convenience.
+- Through duck-typing, JAX arrays can often be used as drop-in replacements of numpy arrays.
+- Unlike numpy arrays, JAX arrays are always immutable.
+
+NumPy provides a well-known, powerful API for working with numerical data.
+For convenience, JAX includes the :mod:`jax.numpy` module, which closely mirrors the
+numpy API and provides easy entry into JAX. Here are some operations you might do in ``numpy``:
```

Comment:
This doc is awesome. 

May I suggest a tiny little thing: `NumPy` <- `numpy`

```suggestion
- JAX provides a Numpy-inspired interface for convenience.
- Through duck-typing, JAX arrays can often be used as drop-in replacements of NumPy arrays.
- Unlike NumPy arrays, JAX arrays are always immutable.

NumPy provides a well-known, powerful API for working with numerical data.
For convenience, JAX includes the :mod:`jax.numpy` module, which closely mirrors the
NumPy API and provides easy entry into JAX. Here are some operations you might do in ``numpy``:
```

([Here](https://jax.readthedocs.io/en/latest/_modules/numpy.html?highlight=numpy#) it's also mostly consistent)

Reply:
Absolutely - thanks!